privacyidea/lib/tokens/radiustoken.py

```diff
@@ -248,34 +485,40 @@ def check_otp(self, otpval, counter=None, window=None, options=None):
                                        NAS_Identifier=nas_identifier.encode('ascii'))
 
             req["User-Password"] = req.PwCrypt(otpval)
-            if "transactionid" in options:
-                req["State"] = str(options.get("transactionid"))
+
+            if radius_state:
+                req["State"] = str(radius_state)
```

Comment:
I think ``radius_state`` should be a bytestring anyway, so we don't need ``str()`` here (and it would break under Python 3)

Reply:
done in 13d94769